### PR TITLE
[lit-next] Add lit-element-starter-{ts,js} templates to monorepo

### PR DESCRIPTION
**Pull request by kevinpschaaf**
Imports the `lit-element-starter-ts` and `lit-element-starter-js` templates into the monorepo (including their git history) as lerna packages, and makes the following changes:
* Upgrades the source to work with `lit-next`
* Aligns the linting and formatting rules/settings & ignores with the monorepo (note this is slightly different than the other packages in that the lint/format commands must be runnable from the package root rather than just the monorepo root). The changes here ensure running format or lint at the monorepo root (as will be done by husky during a commit) or at the package root (as will be done by end users) results in the same output.
* Upgrades to the same version of typescript and tsconfig settings as the monorepo
* Upgrades the test runner from karma to `web-test-runner`
* Upgrades the dev server from `es-dev-server` to `@web/dev-server`

Although we want to maintain `lit-element-starter-ts` and `lit-element-starter-js` in the monorepo for ease of integration testing, the source is exported back out to individual repos as these are [GitHub Template Repositories](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/creating-a-template-repository) with a nice workflow for users to create their own new element repos based on the template. The following commands to export new commits to the monorepo packages to a branch on the template repos (`lit-next` branch shown in example):

```bash
# Export TS template
git remote add lit-element-starter-ts git@github.com:PolymerLabs/lit-element-starter-ts.git
git subtree push --prefix=packages/lit-element-starter-ts/ lit-element-starter-ts lit-next
# Export JS template
git remote add lit-element-starter-ts git@github.com:PolymerLabs/lit-element-starter-ts.git
git subtree push --prefix=packages/lit-element-starter-js/ lit-element-starter-js lit-next
```
(These instructions are added to the top-level README)

### CODE REVIEW

File: README.md
Comment (justinfagnani) [inline README.md:74]:
```suggestion
git remote add lit-element-starter-js git@github.com:PolymerLabs/lit-element-starter-js.git
```

File: packages/lit-element-starter-js/README.md
Comment (justinfagnani) [inline packages/lit-element-starter-js/README.md:15]: Does it still use Karma?

File: packages/lit-element-starter-js/README.md
Comment (justinfagnani) [inline packages/lit-element-starter-js/README.md:25]: Does it still use es-dev-server?

File: packages/lit-element-starter-js/custom-elements.json
Comment (justinfagnani) [inline packages/lit-element-starter-js/custom-elements.json:1]: shouldn't this not be checked in? Is it in the original project?

File: packages/lit-element-starter-js/docs/_README/index.html
Comment (justinfagnani) [inline packages/lit-element-starter-js/docs/_README/index.html:1]: Hmm... I think the idea with the `_` prefix on _README.md was that it would be ignored by eleventy. Ideally we don't generate this.

File: packages/lit-element-starter-js/package.json
Comment (justinfagnani) [inline packages/lit-element-starter-js/package.json:35]: pre.2 now? Or `@next-major`?

File: packages/lit-element-starter-js/package.json
Comment (justinfagnani) [inline packages/lit-element-starter-js/package.json:57]: I wonder what uses lit-html directly, but only in dev? I think this could be removed

File: packages/lit-element-starter-js/package.json
Comment (justinfagnani) [inline packages/lit-element-starter-js/package.json:38]: some of these are outdated. We can update later

File: packages/lit-element-starter-js/rollup.config.js
Comment (justinfagnani) [inline packages/lit-element-starter-js/rollup.config.js:40]: I think we should add a comment here to highlight this behavior. It could break thing in the docs site if devs access `__` prefixed names somehow? And we might not need it anymore if we ensure the site uses the default build of Lit*

**Review by justinfagnani — CHANGES_REQUESTED** [review]
First pass on the JS version

File: .eslintignore
Comment (sorvell) [inline .eslintignore:52]: Is this needed since your'e ignoring the whole folder?

File: .eslintignore
Comment (sorvell) [inline .eslintignore:63]: Is this needed since your'e ignoring the whole folder?

File: packages/lit-element-starter-ts/src/my-element.ts
Comment (sorvell) [inline packages/lit-element-starter-ts/src/my-element.ts:15]: Add TODO to change to `lit` when it's available. Also do in the JS version.

**Review by sorvell — APPROVED** [review]
Hard to closely review but basically looks OK with a couple nits.

**Comment by kevinpschaaf** [comment]
@google-cla Flagged commits were CLA-approved on repos they were imported from ([lit-element-starter-ts ](https://github.com/PolymerLabs/lit-element-starter-ts/pulls?q=is%3Apr+is%3Aclosed), [lit-element-starter-js ](https://github.com/PolymerLabs/lit-element-starter-js/pulls?q=is%3Apr+is%3Aclosed)).

File: packages/lit-element-starter-js/custom-elements.json
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/custom-elements.json:1]: Yeah, it's checked in in the original templates: https://github.com/PolymerLabs/lit-element-starter-ts/blob/master/custom-elements.json. Not sure what the thinking was there...? The generated docs are also checked in, but that's to use github pages's easy sub-folder publishing setting right?

File: .eslintignore
Comment (kevinpschaaf) [reply .eslintignore:52]: This is a generated file (by `npm run ignore-sync`); since `packages/lit-element-starter-ts/.eslintignore` has a (package) top-level `node_modules/*` ignore, that gets turned into this at the monorepo root. It's fine I think.

File: .eslintignore
Comment (kevinpschaaf) [reply .eslintignore:63]: Same comment above.

File: packages/lit-element-starter-js/custom-elements.json
Comment (justinfagnani) [reply packages/lit-element-starter-js/custom-elements.json:1]: Yeah, generated docs are checked in for github pages. I don't think this should be there since it's not used in the static site.

File: README.md
Comment (kevinpschaaf) [reply README.md]: Done

File: packages/lit-element-starter-js/README.md
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/README.md]: Updated to `@web/test-runner`

File: packages/lit-element-starter-js/README.md
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/README.md]: Updated to `@web/dev-server`

File: packages/lit-element-starter-js/package.json
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/package.json]: Done (`next-major`)

File: packages/lit-element-starter-js/package.json
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/package.json]: Removed

File: packages/lit-element-starter-ts/src/my-element.ts
Comment (kevinpschaaf) [reply packages/lit-element-starter-ts/src/my-element.ts:17]: Done.

File: packages/lit-element-starter-js/docs/_README/index.html
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/docs/_README/index.html:1]: Done. Added an `.eleventyignore` to exclude.

File: packages/lit-element-starter-js/rollup.config.js
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/rollup.config.js:40]: It seems reasonable to have this for mangling properties of e.g. the `MyElement` class. But we should probably align this with the single `_` we do in core.... wdyt?

**Comment by kevinpschaaf** [comment]
@sorvell @justinfagnani Updated, PTAL

**Comment by arthurevans** [comment]
@googlebot I consent.

**Comment by arthurevans** [comment]
Looks like @justinfagnani and @bicknellr are the other committers who probably need to talk to googlebot.

**Comment by bicknellr** [comment]
@googlebot I consent.

**Comment by kevinpschaaf** [comment]
@arthurevans https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2FPolymer%2Flit-html%2Fpull%2F1518 indicates it's the external contributors. Since the CLA bot was set up on the original repos and all PR's were CLA-bot approved, I'm assuming it's fine here and we don't need to bug them to consent again.

**Comment by justinfagnani** [comment]
@googlebot I consent.

File: packages/lit-element-starter-ts/dev/index.html
Comment (sorvell) [inline packages/lit-element-starter-ts/dev/index.html:7]: Add wc polyfills and `polyfill-support`

File: packages/lit-element-starter-js/dev/index.html
Comment (sorvell) [inline packages/lit-element-starter-js/dev/index.html:7]: Add wc polyfills and `polyfill-support`

File: packages/lit-element-starter-ts/dev/index.html
Comment (kevinpschaaf) [reply packages/lit-element-starter-ts/dev/index.html:9]: Done

File: packages/lit-element-starter-js/dev/index.html
Comment (kevinpschaaf) [reply packages/lit-element-starter-js/dev/index.html:9]: Done

**Comment by kevinpschaaf** [comment]
@justinfagnani @sorvell PTAL: added polyfills support and switched to using `lit` package.

**Comment by kevinpschaaf** [comment]
Note that although I updated to use the `lit` package, I couldn't remove `lit-element` from the dependencies because `@open-wc/testing` brings in `@open-wc/scoped-elements` which has a dependency on lit-element but doesn't declare it. Filed https://github.com/open-wc/open-wc/issues/2085 on that.

**Review by sorvell — APPROVED** [review]

**Review by justinfagnani — APPROVED** [review]

**Comment by kevinpschaaf** [comment]
This code will be merged in https://github.com/Polymer/lit-html/pull/1576, which has the same code but starting from a `git subtree add` into the `lit-starter-ts` and `lit-starter-js` folders (renaming the folder after the subtree add seems to have broken the ability to subtree export).

**Pull request closed without merging**
